### ISSUE TITLE
Extended Keyboard Shortcuts

### ISSUE THREAD

**Issue by sabjorn** [sabjorn/BandcampEnhancementSuite#57]
`,` and `.` jump forward/backward 1s

**Comment by tadasmajeris** [comment]
would love a way to set custom keyboard shortcuts. was used to Firefox extension which had A <- and D -> for prev and next track. and skipping to the middle of the track with a mouse. faster digging :P

**Comment by sabjorn** [comment]
@tadasmajeris sorry for the delay. I fully agree -- custom shortcuts is definitely a must have. I think I have most of the infrastructure to support it but I just haven't had any time to work on this yet. However, I will make an effort to try to get this going.

**Comment by sabjorn** [comment]
might be possible to just use "Commands":

https://developer.chrome.com/docs/extensions/reference/commands/

this should allow users to re-bind keyboard commands without having to build my own UI for it

**Comment by npp1993** [comment]
adding keyboard shortcuts to the wishlist page would be amazing.  namely:
- arrow keys to seek thru tracks, same as regular shortcuts on track/album pages
- shortcut to add current track to cart
- shortcut to remove current track from wishlist

**Comment by sabjorn** [comment]
@npp1993 I am currently working on #179 which will be the start of being able to add tracks to your cart. The first pass will only be for album/track pages but I think adding to Wishlist in the future should be possible.

for keyboard shortcuts -- I currently do not have a plan to add any seeking to the Wishlist/Collection player. I have experimented a bit with just replacing all the players on the site with a single unified one -- which will also allow me to replace the "preview" buttons on label pages with a single playlist system. Which also means all keyboard shortcuts will be shared between the different pages.

the shortcut for removing from the wishlist is definitely possible but also will be a while.

**Comment by npp1993** [comment]
Awesome, I appreciate the additional context.  

The reason I request these features is that I use the wishlist as a "maybe going to buy" list, which I then periodically go through and listen to.  As I do this I remove or add each item to my cart, which I then purchase after completing this process.  I imagine other people do the same thing, so enhancements to the app that speed up this workflow would be really nice.

Many thanks for making this project available to the public for free!

**Comment by sabjorn** [comment]
@npp1993 I feel you on that. It honestly really frustrates me that there isn't this workflow just built into Bandcamp. It is pretty clear that the design of the website isn't really tailored to people like us.

My hope is that when I re-write the audio player you will be able to just load your entire wishlist as a palylist. And with this I also want to enable "playlists" so that you can sort your wishlist into different sections. 
 
Also -- thank you for sponsoring! I have another project that is secret. Send me an email (it's in my github profile) and I can send you an invite if you're interested. I can't give any more details here but you might dig it.

**Comment by npp1993** [comment]
Agreed.  A few small tweaks on their end, and the browsing experience could be so much better.  At least we have BES to get us closer to that ideal experience!

Those planned changes sound awesome.  Love the idea of being able to divite the wishlist into playlists.

Very happy to sponsor!  I evangelize about this tool to my like-minded friends.  DM'd as well

**Comment by sabjorn** [comment]
chrome://extensions/shortcuts

**Comment by sabjorn** [comment]
closed with #210
